### PR TITLE
fix/CICD ~ cleanup and refactor with multiple associated fixes

### PR DESCRIPTION
**Pull request by rivy**
The plan is for this commit set to get CI to back to green (✔️).
I'll convert from draft when the set builds/tests cleanly.

### CODE REVIEW

**Comment by rivy** [comment]
All checks/tests are now passing.

File: src/uu/touch/src/touch.rs [hunk 1]
```diff
@@ -84,13 +84,9 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
...
         (Some(reference), Some(date)) => {
             let (atime, mtime) = stat(Path::new(reference), !matches.get_flag(options::NO_DEREF))?;
-            if let Ok(offset) = humantime_to_duration::from_str(date) {
-                let mut seconds = offset.whole_seconds();
-                let mut nanos = offset.subsec_nanoseconds();
-                if nanos < 0 {
-                    nanos += 1_000_000_000;
-                    seconds -= 1;
-                }
+            if let Ok(offset) = parse_datetime::from_str(date) {
```
Comment (sylvestre) [inline src/uu/touch/src/touch.rs:87]: please move this change into a different PR

File: .taplo.toml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# `taplo` configuration
```
Comment (sylvestre) [inline .taplo.toml:1]: i would rather keep the defaults of taplo

File: deny.toml [hunk 3]
```diff
@@ -27,7 +27,7 @@ allow = [
   "BSD-2-Clause-FreeBSD",
   "BSD-3-Clause",
   "CC0-1.0",
-  "MPL-2.0",              # XXX considered copyleft?
+  # "MPL-2.0",              # * unused # ToDO?: considered copyleft? remove?
```
Comment (sylvestre) [inline deny.toml:30]: please just remove it :)

File: .taplo.toml (hunk 2 above)
Comment (rivy) [reply .taplo.toml:1]: It splits many of the dependency lines in the `Cargo.toml` files making it harder to keep them organized.

File: src/uu/touch/src/touch.rs (hunk 1 above)
Comment (rivy) [reply src/uu/touch/src/touch.rs:87]: I can, but the point was to get "green" and stop all the failed CI builds.

**Comment by rivy** [comment]
I'm rechecking the Android CI step; it looks like it timed-out waiting for the emulator.

File: .taplo.toml (hunk 2 above)
Comment (sylvestre) [reply .taplo.toml:1]: Do you have examples of that?

For example, this one seems unrelated:
array_auto_collapse = false

File: src/uu/touch/src/touch.rs (hunk 1 above)
Comment (sylvestre) [reply src/uu/touch/src/touch.rs:87]: It doesn't really matter IMHO

File: Cargo.toml [hunk 4]
```diff
@@ -315,14 +320,18 @@ rstest = "0.17.0"
...
 signal-hook = "0.3.15"
+sm3 = "0.4.2"
 smallvec = { version = "1.10", features = ["union"] }
 tempfile = "3.6.0"
 term_grid = "0.1.5"
 terminal_size = "0.2.6"
 textwrap = { version = "0.16.0", features = ["terminal_size"] }
 thiserror = "1.0"
-time = { version = "0.3" }
+time = { version = "=0.3.20" } ## maint: [2023-06-11; rivy] pinned to maintain MinSRV 1.64.0
```
Comment (rivy) [inline Cargo.toml:334]: @sylvestre , this end-of-line comment gets blown out to starting at 100+ character position without the `align_comments = false`.

Comment (sylvestre) [reply Cargo.toml:334]: you can probably make the comment shorter. remove the "maint" and the "maintain" 
or move it the line above

File: .taplo.toml (hunk 2 above)
Comment (rivy) [reply .taplo.toml:1]: I added them iteratively as formatting problems cropped up. 
I don't remember the details for the need for that line.
It might have been needed before the final format was completed, and redundant now.
I'll remove that line it locally and see if anything crops back up.

File: Cargo.toml (hunk 4 above)
Comment (rivy) [reply Cargo.toml:334]: IMO, it's better located on the line that it refers to so that they don't get separated with future change.
Ultimately, it'll be removable (and should be removed along with the `=` pin) when MinSRV increases.
And, no matter the length, if it's on the same line, `taplo` is pushing it to the right side, with an extremely wide area of white space between it and the code it refers too.

File: .taplo.toml (hunk 2 above)
Comment (rivy) [reply .taplo.toml:1]: All of the configuration tags here are purposeful and help me keep the TOML organized and readable.

**Comment by rivy** [comment]
> Could you please split this PR into smaller? it is a mixing way too many different things in the same PR.
> 
> I don't mind having individual PR being red
> 
> Thanks

Sure, I'll try to refine it and split it.
It's conceptually a whole in that it is fixing the technical debt of allowing multiple failed CI merges.
I'll try to remove any commits that aren't directly contributing to getting CI back into working order.

I get that a failed CI isn't the worst of issues, but it becomes chronic. And then things start slipping through that aren't seen because we think we know why it's failing. It becomes much more messy to fix it.

**Comment by sylvestre** [comment]
most of the changes have been merged: closing
